draft-ietf-httpbis-connect-tcp.md

```diff
@@ -72,6 +72,12 @@ An intermediary MAY merge and split successive DATA capsules, subject to the fol
 * There are no intervening capsules of other types.
 * The order of payload content is preserved.
 
+An END_DATA capsule indicates that sender has finished sending data for this stream.  Its payload is a QUIC varint representing an error code.  A value of 0 means "completed", semantically equivalent to TCP FIN.  Any other value indicates truncated data, corresponding to a TCP RST, timeout, or other error.  In this specification, the only other defined value is 1, meaning "truncated".
+
+A STOP_SENDING_DATA capsule indicates that the sender has begun discarding DATA capsules that it receives on this stream.  Its payload is empty.
```

Comment:
This is going to end up being confusing because it is close to the semantic of QUIC's STOP_SENDING, but not quite.

I understand that you might want to have this capability for HTTP/2 and HTTP/1.1, where our model does not properly decouple send and receive sides of a bidirectional exchange.  This is a function that is already provided by the reset function in each protocol (RST_STREAM in HTTP/2, TCP RST in HTTP/1.1).  

While you might think that it is necessary to cover for those shortcomings, I don't see how that is necessary.  You are  only providing a redundant mechanism/signal.

A positive signal that a stream ended cleanly is useful, but the absence of that signal is sufficient to indicate truncation.

Reply:
I agree with MT, I don't think we need STOP_SENDING_DATA. For h2 and h3, sending RST_STREAM/RESET_STREAM without sending FINAL_DATA is enough to indicate an abortive close. For h1, we have a mechanism (TCP RST / TLS error alert) that's good enough even if it's not possible to implement for everyone.